stan/math/rev/functor/adj_arg.hpp

```diff
@@ -17,23 +17,23 @@ namespace math {
  *  type is passed this class will be empty.
```

Comment:
Replace:

```
Stores the values for the functors passed to adj_jac_vari. Passing a var or
or container of vars will store a pointer to the val. If a non-var
type is passed this class will be empty.
```

with something more descriptive. Something like:

```
This is a utility for optionally storing dynamically allocated variables (`std::vector<T>`, `Eigen::Matrix<T, R, C>`, ...)
on the autodiff stack instead of the heap. The first template argument is the type of the variable to be saved. The
second template argument is a boolean true/false indicating whether to actually allocate and save the variables
on the stack.

By default (if no second argument is specified) the values of the input will be saved to the stack if the input contains
vars, otherwise nothing is saved.

This utility exposes the saved memory both as an `Eigen::Map` and through accessors.
```